lessons/jest.md

```diff
@@ -0,0 +1,89 @@
+# ModernJS - Jest
+## Welcome to Modern JS
+
+## Jest
+In this lesson we'll be learning about [Jest](https://facebook.github.io/jest) which is an easy and powerful tool for testing your javascript projects.
+
+### What is Jest?
+It is a unit testing framework, creating with the idea of being a "zero configuration" testing experience.
+We'll spend less time setting up test runners, assertion libraries, mocking tools and more time writing tests, leading to more stable and healthy code bases. 
+
+### Features
+- Automatically generating snapshots for snapshot testing
+- Running tests in parallel, meaning it's super performant
+- Sandboxing tests and automatically tearing down global state so you don't have to worry about conflicts
+- Code coverage reports
+- Built-in mocking and assertions
+
+## Installing Jest
+Jest is available on npm.
+
+```
+// yarn
+yarn add --dev jest
+
+// npm
+npm i --save-dev jest
+```
+
+### Using Jest with babel
+Jest runs on Node and as a result, if you're using ES2016+ you'll need to transpile your code. A nice side effect of this is that we can also write our tests in ES2016+ too.
+
+*Caveat* - The version of babel from the first lesson is a beta version of babel 7. Jest is not yet compatible with this, so we'll need to include `babel-core@^7.0.0-0` and  `regenerator-runtime` to get the same behaviour in the testing enviroment.
+
+Get the compatible version of `babel-core` and `regenerator-runtime`
+```
+// yarn
+yarn add --dev babel-core@^7.0.0-0 regenerator-runtime
+
+// npm
+npm i --save-dev babel-core@^7.0.0-0 regenerator-runtime
+```
+
+### Configuring Jest
+Out of the box, running Jest will search your project for `<filename>.test.js` and run them against `<filename>.js`. This is great but sometimes we'll need more.
```

Comment:
Does it run it against `<filename>.js` or do you still have to explicitly import it? I feel like that's just a naming convention but I don't know as I tend to use `__tests__` dirs and `<filename>.spec.js`.